src/main/metabase/examples.ts

```diff
@@ -0,0 +1,152 @@
+import { postgresQuote } from '../../shared';
+import { Field, Table } from './types';
+
+type Display = 'table' | 'bar' | 'row' | 'scalar' | 'map'; // Other types TBD.
+
+/** An example query card. */
+type ExampleQuery = {
+  name: string; // Title of card.
+  sql: string; // SQL query.
+  sizeX: number; // Grid of 18 units wide.
+  sizeY: number; // Height of card in units.
+  display: Display;
+  visualizationSettings: Record<string, unknown>; // To be typed later.
+
+  // There's also row/col properties, but we'll make some rectangle
+  // packing algorithm to arrange cards automatically in a section.
+};
+
+/** A section for a group of examples. */
+type ExamplesSection = {
+  title: string | null; // Markdown text as the section heading.
+  queries: ExampleQuery[]; // Cards in section.
+};
+
+type ExampleInfo = {
+  sql: string;
+  name: string;
+};
+
+function lines(...lines: string[]) {
+  return lines.join('\n');
+}
+
+const numberFieldTypes = ['int2', 'int4', 'int8', 'float4', 'float8', 'numeric'];
+
+function rawGroupBySQL(column: string, table: string, displayName: string): ExampleInfo {
+  return {
+    name: `${displayName} by ${column}`,
+    sql: lines(`SELECT ${postgresQuote(column)}`, `FROM ${postgresQuote(table)}`),
+  };
+}
+
+function countDistinctSQL(column: string, table: string): ExampleInfo {
+  return {
+    name: `Distinct ${column}`,
+    sql: lines(`SELECT count(distinct ${postgresQuote(column)}) as distinct_${column}`, `FROM ${postgresQuote(table)}`),
```

Comment:
Do we handle quoting of this? If so, we might get `distinct_"some col"`

Reply:
I think I checked mb reports the table unquoted here, so we should be fine. I'll check again

Reply:
unquoted, but I have to prevent `distinct_some col`, I'll fix this

Reply:
> I'll fix this

Let's fix bugs in a future PR. I want to synchronize and experiment often.